### ISSUE TITLE
nonminimal_bool produces suboptimal and invalid results

### ISSUE THREAD

**Issue by pacak** [rust-lang/rust-clippy#11921]
### Summary

See code

### Lint Name

nonminimal_bool

### Reproducer

I tried this code:

Let's start with this somewhat strange code - note a problem - c is used twice, b is missing.
```rust
pub fn foo1(a: u8, b: u8, c: u8) -> bool {
    !(a < 100 || c < 100 || c < 100)
}
```

I saw this happen:

```
warning: this boolean expression can be simplified
 --> src/lib.rs:2:5
  |
2 |     !(a < 100 || c < 100 || c < 100)
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#nonminimal_bool
  = note: `#[warn(clippy::nonminimal_bool)]` on by default
help: try
  |
2 |     !(a < 100 || c < 100)
  |
2 |     a >= 100 && c >= 100
  |

warning: this boolean expression can be simplified
 --> src/lib.rs:2:6
  |
2 |     !(a < 100 || c < 100 || c < 100)
  |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#nonminimal_bool
help: try
  |
2 |     !!(a >= 100 && c >= 100)
  |      ~~~~~~~~~~~~~~~~~~~~~~~
2 |     !a < 100 || c < 100
  |      ~~~~~~~~~~~~~~~~~~
```

For some reason problem is reported twice and it proposes 4 different solutions. First two are good, but they are missing the point of typo in `b`, third one is technically correct but is less readable (and if you do this substitution - clippy won't try to simplify it further), fourth one is plain wrong.

I expected to see this happen:

Error is reported once with either of first two solutions or both. Bonus points for showing out that a component was used twice.

### Version

```text
rustc 1.74.0 (79e9716c9 2023-11-13)
binary: rustc
commit-hash: 79e9716c980570bfd1f666e3b16ac583f0168962
commit-date: 2023-11-13
host: x86_64-unknown-linux-gnu
release: 1.74.0
LLVM version: 17.0.4
```
